packages/webpack/src/utils/module-federation/share.ts

```diff
@@ -66,14 +76,59 @@ export function shareWorkspaceLibraries(
         (aliases, library) => ({ ...aliases, [library.name]: library.path }),
         {}
       ),
-    getLibraries: (eager?: boolean): Record<string, SharedLibraryConfig> =>
-      pathMappings.reduce(
-        (libraries, library) => ({
+    getLibraries: (eager?: boolean): Record<string, SharedLibraryConfig> => {
+      let pkgJson = null;
+      if (
+        project?.root &&
+        existsSync(
+          joinPathFragments(workspaceRoot, project.root, 'package.json')
+        )
+      ) {
+        pkgJson = readJsonFile(
```

Comment:
This can fail if the project does not have `package.json`. We should check if it exists first.

Reply:
This has already been checked above line 81-85
```
if(projectRoot && existsSync(joinPathFragments(workspaceRoot, projectRoot, 'package.json'))
```